commit d3060fef380723635f2d3bb39ae697cd0fdde30d (HEAD -> togglefloatingcenter)
Author: Rizqi Nur Assyaufi <bandithijo@gmail.com>
Date:   Fri Aug 6 14:37:10 2021 +0800
Subject: [patch][togglefloatingcenter] centered togglefloating window

Default behaviour when togglefoating() is floating from top-left corner.
This patch will allows you to toggle floating window client will be centered
position.

---
 dwm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dwm.c b/dwm.c
index 664c527..5234d4b 100644
--- a/dwm.c
+++ b/dwm.c
@@ -1719,6 +1719,10 @@ togglefloating(const Arg *arg)
    if (selmon->sel->isfloating)
        resize(selmon->sel, selmon->sel->x, selmon->sel->y,
            selmon->sel->w, selmon->sel->h, 0);
+
+    selmon->sel->x = selmon->sel->mon->mx + (selmon->sel->mon->mw - WIDTH(selmon->sel)) / 2;
+    selmon->sel->y = selmon->sel->mon->my + (selmon->sel->mon->mh - HEIGHT(selmon->sel)) / 2;
+
    arrange(selmon);
 }
--
2.31.1
